execution_engine/src/shared/account/associated_keys.rs

```diff
@@ -328,4 +334,21 @@ mod tests {
             .unwrap();
         bytesrepr::test_serialization_roundtrip(&keys);
     }
+
+    #[test]
+    fn should_not_panic_deserializing_malicious_data() {
```

Comment:
Feeling mildly uncomfortable here.  Agree that we shouldn't panic, but can somebody game this?